daft/udf/udf_v2.py

```diff
@@ -170,6 +175,20 @@ def _derive_function_name(self) -> str:
         else:
             return f"{qual_name}-{uuid.uuid4()}"
 
+    @staticmethod
+    def _get_input_dtypes(fn: Callable[..., Any]) -> dict[str, DataType]:
+        type_hints = get_type_hints(fn)
+        sig = inspect.signature(fn)
+        input_dtypes = {}
+        for param_name in sig.parameters:
+            if param_name == "self":
+                continue
+            if param_name not in type_hints:
+                input_dtypes[param_name] = DataType.python()
+            else:
+                input_dtypes[param_name] = DataType._infer(type_hints[param_name])
```

Comment:
Small nit but we can use `DataType.infer_from_type` here instead, it is more specific